Controller/CRUDController.php

```diff
@@ -265,6 +265,9 @@ public function editAction($id = null)
 
             // persist if the form was valid and if in preview mode the preview was approved
             if ($isFormValid && (!$this->isInPreviewMode() || $this->isPreviewApproved())) {
+                $object = $form->getData();
+                $this->admin->setSubject($object);
```

Comment:
The reason is a newly updated `$object` should be passed into `$object = $this->admin->update($object);` It was done for purpose. If form somehow changes object and it is no longer the same instance then admin won't get that changed object in `$this->admin->update($object)`.